client/my-sites/email/form/mailboxes/components/new-mailbox-list/use-get-existing-mailbox-names.ts

```diff
@@ -0,0 +1,51 @@
+import { useSelector } from 'react-redux';
+import { useGetEmailAccountsQuery } from 'calypso/data/emails/use-get-email-accounts-query';
+import {
+	EMAIL_ACCOUNT_TYPE_GOOGLE_WORKSPACE,
+	EMAIL_ACCOUNT_TYPE_GSUITE,
+	EMAIL_ACCOUNT_TYPE_TITAN_MAIL,
+	EMAIL_ACCOUNT_TYPE_TITAN_MAIL_EXTERNAL,
+} from 'calypso/lib/emails/email-provider-constants';
+import { EmailProvider } from 'calypso/my-sites/email/form/mailboxes/types';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+interface ExistingMailAccount {
+	account_type: string;
+	emails: {
+		email_type: string;
+		mailbox: string;
+	}[];
+}
+
+const mapProviderToEmailAccountTypes = ( provider: EmailProvider ): string[] =>
+	( {
+		[ EmailProvider.Titan ]: [
+			EMAIL_ACCOUNT_TYPE_TITAN_MAIL,
+			EMAIL_ACCOUNT_TYPE_TITAN_MAIL_EXTERNAL,
+		],
+		[ EmailProvider.Google ]: [ EMAIL_ACCOUNT_TYPE_GOOGLE_WORKSPACE, EMAIL_ACCOUNT_TYPE_GSUITE ],
+	}[ provider ] );
+
+const useGetExistingMailboxNames = (
+	provider: EmailProvider,
+	selectedDomainName: string
+): string[] => {
+	const selectedSiteId = useSelector( getSelectedSiteId );
+	const {
+		data: { accounts } = {},
+		error,
+		isLoading,
+	} = useGetEmailAccountsQuery( selectedSiteId as number, selectedDomainName );
+
+	if ( error || isLoading || ! accounts ) {
+		return [];
+	}
+
+	const emailAccountTypes = mapProviderToEmailAccountTypes( provider );
```

Comment:
What about simply naming this function `getEmailAccountTypes()`. `mapProviderToEmailAccountTypes` seems a bit convoluted to me.